src/app/member/components/MemberEditPage.tsx

```diff
@@ -0,0 +1,369 @@
+    )
+    .max(
+      new Date(),
+      "You can not suitable for this job because your are under age"
+    ),
+  cnic: z.string().regex(cnicValidator, "CNIC is not valid"),
+  // shift: z.nativeEnum(ShiftTiming, {
+  //   errorMap: () => {
+  //     return { message: "Please select your user type" };
+  //   },
+  // }),
+  // address: z
+  //   .string({ message: "Address is required" })
+  //   .min(10, "Name should have at least 10 characters"),
+  role: z.nativeEnum(UserRole, {
+    errorMap: () => {
+      return { message: "Please select your user type" };
+    },
+  }),
+  // ehad_start_date: z
+  //   .date()
+  //   .min(addDays(new Date(), 30), "Minmum Ehad Duration is one Month"),
+});
+
+export const MemberEditPage = () => {
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      name: "",
+      phone: "",
+      date_of_birth: new Date(),
+      cnic: "",
+      // shift: undefined,
+      // address: "",
+      role: undefined,
+      // ehad_start_date: new Date(),
+    },
+    mode: "onBlur",
+  });
+  const handleSubmittion = async (values: z.infer<typeof formSchema>) => {
+    // const data = await onSubmit(values);
+    // if (data) {
+    //   console.log("Data received", data);
+    // }
+    console.log("values", values);
+    formSchema.parse(values);
+    form.reset();
+  };
+  const {
+    formState: { errors },
+  } = form;
+  return (
+    <FormWrapper>
+      <Form {...form}>
+        <form
+          onSubmit={(e) => {
+            e.preventDefault();
+            handleSubmittion(form.getValues());
+          }}
+          className="space-y-4"
+          data-testid="form-valid"
+        >
+          <h4 className="text-xl text-black text-bold">Information</h4>
+          <FormField
+            control={form.control}
+            name="name"
+            render={({ field }) => (
+              <FormItem className="flex flex-col">
+                <Label>Name</Label>
+
+                <Input
+                  placeholder="Enter Your Name"
+                  {...field}
+                  data-testid="name"
+                  name="name"
+                  className={
+                    errors?.name &&
+                    "border-red-500 border focus-visible:ring-0 "
+                  }
+                />
+
+                <FormMessage data-testid="name-error" />
+              </FormItem>
+            )}
+          />
+          <FormField
+            control={form.control}
+            name="phone"
+            render={({ field }) => (
+              <FormItem className="flex flex-col">
+                <Label>Phone number</Label>
+                <FormControl>
+                  <Input
+                    placeholder="0300-0000000"
+                    {...field}
+                    data-testid="phone"
+                    className={
+                      errors?.phone &&
```

Comment:
Let's add an `hasError` field to input and make this generic.

Reply:
fixed

Reply:
fixed